### PR TITLE
feature/SOF-3635

### PR DESCRIPTION
**Pull request by mohammadidinani**

### CODE REVIEW

File: src/wave.js [hunk 1]
```diff
@@ -73,14 +73,32 @@ class WaveBase {
...
-        this.camera.position.copy(new TV3(...initialPosition));
-        this.camera.up = new TV3(0, 0, 1);
-        this.camera.lookAt(new TV3(0, 0, 0));
-        this.scene.add(this.camera);
+        this.orthographicCamera = new THREE.OrthographicCamera(-10 * this.ASPECT, 10 * this.ASPECT, 10, -10, 1, 1000);
+        this.orthographicCamera.name = "OrthographicCamera";
+        this.orthographicCamera.position.copy(new TV3(...initialPosition));
+        this.orthographicCamera.up = new TV3(0, 0, 1);
+        this.orthographicCamera.lookAt(new TV3(0, 0, 0));
+        this.scene.add(this.orthographicCamera);
```
Comment (timurbazhirov) [inline src/wave.js]: I'd move this to `initCamera` and call that in constructor - shall we?

File: src/wave.js [hunk 2]
```diff
@@ -73,14 +73,32 @@ class WaveBase {
...
+    }
+
+    initPerspectiveCamera() {
+        const initialPosition = [-50, 0, 10];
```
Comment (timurbazhirov) [inline src/wave.js]: Also, we might be using this position in multiple places - move to settings?

File: src/wave.js (hunk 1 above)
Comment (mohammadidinani) [reply src/wave.js]: `initCamera` is divided into two parts, `initOrthographicCamera ` and `initPerspectiveCamera ` both called in the constructor.

File: src/wave.js (hunk 2 above)
Comment (mohammadidinani) [reply src/wave.js]: Done.

File: src/wave.js (hunk 1 above)
Comment (timurbazhirov) [reply src/wave.js]: Well, that the confusing part - we only have one camera active in the scene (line 92) and it happens in one of inits but not the other - let's isolate that line to:

```js
initCameras() {
        this.initPerspectiveCamera();
        this.initOrthographicCamera();
        this.camera = this.perspectiveCamera;
}
```
and call `initCameras` in constructor?

also lines 79-81 and 88-90 are similar - let's move to `setCameraPosition(camera)` and also have 1, 0, 0 and 0, 0, 0 set in settings?

Comment (mohammadidinani) [reply src/wave.js]: Done.

File: src/wave.js [hunk 3]
```diff
@@ -73,14 +72,29 @@ class WaveBase {
...
 
-    initCamera() {
-        const initialPosition = [-50, 0, 10];
-        this.camera = new THREE.PerspectiveCamera(20, this.ASPECT, 1, 20000);
-        this.camera.name = "PerspectiveCamera";
-        this.camera.position.copy(new TV3(...initialPosition));
-        this.camera.up = new TV3(0, 0, 1);
-        this.camera.lookAt(new TV3(0, 0, 0));
-        this.scene.add(this.camera);
+    addCameraToScene(type, ...args) {
```
Comment (timurbazhirov) [inline src/wave.js]: Needs a docstring with params

Comment (mohammadidinani) [reply src/wave.js]: Added in other PRs.

**Pull request merged**